subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestFramework.java

```diff
@@ -50,24 +47,21 @@
     private final TestNGOptions options;
     private final TestNGDetector detector;
     private final DefaultTestFilter filter;
-    private final ObjectFactory objects;
-    private final String testTaskPath;
-    private final FileCollection testTaskClasspath;
     private final Factory<File> testTaskTemporaryDir;
-    private transient ClassLoader testClassLoader;
 
     @UsedByScanPlugin("test-retry")
     public TestNGTestFramework(final Test testTask, FileCollection classpath, DefaultTestFilter filter, ObjectFactory objects) {
         this.filter = filter;
-        this.objects = objects;
-        this.testTaskPath = testTask.getPath();
-        this.testTaskClasspath = classpath;
         this.testTaskTemporaryDir = testTask.getTemporaryDirFactory();
         options = objects.newInstance(TestNGOptions.class);
         conventionMapOutputDirectory(options, testTask.getReports().getHtml());
         detector = new TestNGDetector(new ClassFileExtractionManager(testTask.getTemporaryDirFactory()));
     }
 
+    public TestNGTestFramework(final Test testTask, DefaultTestFilter filter) {
```

Comment:
Can you update the other references to use the new constructor too?

Reply:
Will do.

Reply:
@big-guy and I agreed to let this stay as-is for 7.3